### PR TITLE
Build slepc

### LINKED ISSUE #1099: Add SLEPc for eigenvalue solvers
This should be fairly straightforward now that we build with PETSc. https://slepc.upv.es/

### PR DESCRIPTION
**Pull request by chapman39**
This PR rebuilds Serac TPLs and adds SLEPc as an additional TPL. Leveraging SLEPc will help Serac solve eigenvalue problems.

- [x] Add a simple SLEPc smoke test
- [x] Add SLEPc as an additional TPL Fixes #1099
- [x] Rebuild TPLs

### CODE REVIEW

File: cmake/thirdparty/FindSLEPc.cmake [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# Copyright (c) 2024, Lawrence Livermore National Security, LLC. All rights
+# reserved.  LLNL-CODE-856946
+
+# OFFICIAL USE ONLY This work was produced at the Lawrence Livermore National
+# Laboratory (LLNL) under contract no. DE-AC52-07NA27344 (Contract 44) between
+# the U.S. Department of Energy (DOE) and Lawrence Livermore National Security,
+# LLC (LLNS) for the operation of LLNL.  See license for disclaimers, notice of
+# U.S. Government Rights and license terms and conditions.
```
Comment (white238) [inline cmake/thirdparty/FindSLEPc.cmake]:
```suggestion
# Copyright (c) 2019-2024, Lawrence Livermore National Security, LLC and
# other Serac Project Developers. See the top-level LICENSE file for
# details.
#
# SPDX-License-Identifier: (BSD-3-Clause)

```

Comment (white238) [reply cmake/thirdparty/FindSLEPc.cmake]: Unless this is a direct copy/paste from another code, it should have our copyright. If it did, then we should provide attribution.

File: cmake/thirdparty/FindSLEPc.cmake [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  if (EXISTS ${_path})
+    list(APPEND _slepc_link_libs ${_path})
+  endif()
+endforeach()
+
+blt_list_remove_duplicates(TO _slepc_link_libs)
+set_target_properties(PkgConfig::SLEPC PROPERTIES INTERFACE_LINK_LIBRARIES "${_slepc_link_libs}")
+
+# Add arpack include directories
+target_include_directories(PkgConfig::SLEPC INTERFACE ${ARPACK_DIR}/include)
```
Comment (white238) [inline cmake/thirdparty/FindSLEPc.cmake]:
```suggestion
target_include_directories(PkgConfig::SLEPC SYSTEM INTERFACE ${ARPACK_DIR}/include)
```

File: cmake/thirdparty/SetupSeracThirdParty.cmake [hunk 3]
```diff
@@ -336,6 +335,19 @@ if (NOT SERAC_THIRD_PARTY_LIBRARIES_FOUND)
         set(PETSC_FOUND FALSE)
     endif()
 
+    #------------------------------------------------------------------------------
+    # SLEPC
+    #------------------------------------------------------------------------------
+    if(SLEPC_DIR)
+        serac_assert_is_directory(DIR_VARIABLE SLEPC_DIR)
+        include(${CMAKE_CURRENT_LIST_DIR}/FindSLEPc.cmake)
+        message(STATUS "SLEPc support is ON")
```
Comment (white238) [inline cmake/thirdparty/SetupSeracThirdParty.cmake:347]:
```suggestion
        serac_assert_find_succeeded(PROJECT_NAME SLEPC
                            TARGET PkgConfig::SLEPC
                            DIR_VARIABLE SLEPC_DIR)
        message(STATUS "SLEPc support is ON")
```

I added this helper function right before this PR.

Comment (white238) [reply cmake/thirdparty/SetupSeracThirdParty.cmake:347]: Please fix the indentation... its hard in Github to get right.

**Review by white238 — APPROVED** [review]

**Review by btalamini — APPROVED** [review]

File: cmake/thirdparty/FindSLEPc.cmake (hunk 1 above)
Comment (chapman39) [reply cmake/thirdparty/FindSLEPc.cmake]: Hm, I originally wrote it for LiDO. Should I just use Serac's?

**Pull request merged**